app/services/shops_list_service.rb

```diff
@@ -2,7 +2,10 @@
 
 class ShopsListService
   def open_shops
-    shops_list.ready_for_checkout.all
+    shops_list.
+      ready_for_checkout.
+      distributors_with_active_order_cycles.
+      all
```

Comment:
You are showing nice TDD here. While I find that super helpful in development, I think that the spec and the changed code can be in the same commit. Then a reviewer can see straight away which spec is passing for this new code.


Another guideline is to make specs always pass in the commit history. This enables us to run `git bisect` on a failing spec to see when it started failing. So if you want to commit your spec first, and that's totally legit as well, then you need to declare the spec pending first, so that it doesn't fail. Then when you implement the code, you remove the pending to make it pass properly. That way you clearly mark the spec as now fulfilled with your code change. Sometimes you need several commits to actually make a spec pass and this way you are transparent about when it's done.

Does that make sense to you?

Reply:
Yes that makes sense - I must say I don't often use `pending` but it definitely makes sense for cases like this